js/src/services/vue.service.js

```diff
@@ -0,0 +1,103 @@
+import Vue from 'vue';
+import atkInlineEdit from '../components/inline-edit.component';
+import itemSearch from '../components/item-search.component';
+
+
+let atkComponents = {
+  'atk-inline-edit' : atkInlineEdit,
+  'atk-item-search' : itemSearch,
+};
+
+/**
+ * Singleton class
+ * Create Vue component.
+ */
+class VueService {
+
+  static getInstance() {
+    return this.instance;
+  }
+
+  constructor() {
+    if(!VueService.instance){
+      this.vues = [];
+      this.eventBus = new Vue();
+      VueService.instance = this;
+    }
+    return VueService.instance;
+  }
+
+  /**
+   * Created a Vue component and add it to the vues array.
```

Comment:
description is unclear.  Does that mean that other than those existing 2 components others would need to invoke this method to register themselves? is the name "createAtkVue" miss-leading? maybe not, just raising a question.

Reply:
Yes. Component defined externally would have to use this method to create the Vue instance.